packages/@react-aria/button/src/useButton.ts

```diff
@@ -31,7 +38,62 @@ export interface ButtonAria {
+function useButton(props: AriaButtonProps<'option'>, ref: RefObject<HTMLOptionElement>): ButtonAria<HTMLAttributes<HTMLOptionElement>>;
+function useButton(props: AriaButtonProps<'output'>, ref: RefObject<HTMLOutputElement>): ButtonAria<HTMLAttributes<HTMLOutputElement>>;
+function useButton(props: AriaButtonProps<'p'>, ref: RefObject<HTMLParagraphElement>): ButtonAria<HTMLAttributes<HTMLParagraphElement>>;
+function useButton(props: AriaButtonProps<'picture'>, ref: RefObject<HTMLPictureElement>): ButtonAria<HTMLAttributes<HTMLPictureElement>>;
+function useButton(props: AriaButtonProps<'pre'>, ref: RefObject<HTMLPreElement>): ButtonAria<HTMLAttributes<HTMLPreElement>>;
+function useButton(props: AriaButtonProps<'progress'>, ref: RefObject<HTMLProgressElement>): ButtonAria<HTMLAttributes<HTMLProgressElement>>;
+function useButton(props: AriaButtonProps<'select'>, ref: RefObject<HTMLSelectElement>): ButtonAria<HTMLAttributes<HTMLSelectElement>>;
+function useButton(props: AriaButtonProps<'slot'>, ref: RefObject<HTMLSlotElement>): ButtonAria<HTMLAttributes<HTMLSlotElement>>;
+function useButton(props: AriaButtonProps<'source'>, ref: RefObject<HTMLSourceElement>): ButtonAria<HTMLAttributes<HTMLSourceElement>>;
+function useButton(props: AriaButtonProps<'span'>, ref: RefObject<HTMLSpanElement>): ButtonAria<HTMLAttributes<HTMLSpanElement>>;
+function useButton(props: AriaButtonProps<'svg'>, ref: RefObject<SVGElement>): ButtonAria<HTMLAttributes<SVGElement>>;
+function useButton(props: AriaButtonProps<'table'>, ref: RefObject<HTMLTableElement>): ButtonAria<HTMLAttributes<HTMLTableElement>>;
+function useButton(props: AriaButtonProps<'td'>, ref: RefObject<HTMLTableDataCellElement>): ButtonAria<HTMLAttributes<HTMLTableDataCellElement>>;
+function useButton(props: AriaButtonProps<'template'>, ref: RefObject<HTMLTemplateElement>): ButtonAria<HTMLAttributes<HTMLTemplateElement>>;
+function useButton(props: AriaButtonProps<'textarea'>, ref: RefObject<HTMLTextAreaElement>): ButtonAria<HTMLAttributes<HTMLTextAreaElement>>;
+function useButton(props: AriaButtonProps<'tfoot'>, ref: RefObject<HTMLTableSectionElement>): ButtonAria<HTMLAttributes<HTMLTableSectionElement>>;
+function useButton(props: AriaButtonProps<'th'>, ref: RefObject<HTMLTableHeaderCellElement>): ButtonAria<HTMLAttributes<HTMLTableHeaderCellElement>>;
+function useButton(props: AriaButtonProps<'thead'>, ref: RefObject<HTMLTableSectionElement>): ButtonAria<HTMLAttributes<HTMLTableSectionElement>>;
+function useButton(props: AriaButtonProps<'time'>, ref: RefObject<HTMLTimeElement>): ButtonAria<HTMLAttributes<HTMLTimeElement>>;
+function useButton(props: AriaButtonProps<'title'>, ref: RefObject<HTMLTitleElement>): ButtonAria<HTMLAttributes<HTMLTitleElement>>;
+function useButton(props: AriaButtonProps<'ul'>, ref: RefObject<HTMLUListElement>): ButtonAria<HTMLAttributes<HTMLUListElement>>;
```

Comment:
I think we should probably restrict these to elements that would make sense as a button. Maybe `button`, `a`, `input`, `div` and `span`?